### ISSUE TITLE
Out of sockets on 3rd Request

### ISSUE THREAD

**Issue by jfurcean** [adafruit/Adafruit_CircuitPython_Requests#73]
The [MagTag Google Calendar Event Display Learn Guide](https://learn.adafruit.com/magtag-google-calendar-event-display/getting-the-date-time) is crashing with the following trace using CircuitPython 6.2.0-beta.2 and the latest libraries.

```
"Traceback (most recent call last):
  File "code.py", line 252, in <module>
  File "code.py", line 129, in get_calendar_events
  File "adafruit_requests.py", line 598, in get
  File "adafruit_requests.py", line 559, in request
  File "adafruit_requests.py", line 436, in _get_socket
  File "adafruit_requests.py", line 433, in _get_socket
RuntimeError: Out of sockets"
```

**Comment by jfurcean** [comment]
I did a little bit of troubleshooting and I think the oauth request socket is being left open. I am not sure if this is a `adafruit_requests` issue or an `adafruit_oauth2`.

**Comment by jfurcean** [comment]
This looks like a failure to close sockets. The below example crashes on the third request.

```python
from adafruit_magtag.magtag import MagTag

magtag = MagTag()
magtag.network.connect()

requests = magtag.network.requests

urls = ["https://wired.com/", "https://www.test.com/","https://apple.com/"]

for url in urls:
    print(url)
    resp = requests.get(url)

    resp.close()
```

```
Traceback (most recent call last):
  File "code.py", line 12, in <module>
  File "adafruit_requests.py", line 616, in get
  File "adafruit_requests.py", line 575, in request
  File "adafruit_requests.py", line 445, in _get_socket
  File "adafruit_requests.py", line 441, in _get_socket
RuntimeError: Out of sockets
```

**Comment by anecdata** [comment]
Just for completeness, it's unrelated to MagTag. This also gets `Out of sockets`:
```python
import wifi
import socketpool
import ssl
import espidf
import adafruit_requests
from secrets import secrets

wifi.radio.connect(secrets['ssid'], secrets['password'])
pool = socketpool.SocketPool(wifi.radio)
requests = adafruit_requests.Session(pool, ssl.create_default_context())

urls = ["https://wired.com/", "https://www.test.com/", "https://apple.com/"]

for url in urls:
    print(url)
    resp = requests.get(url)
    resp.close()
```

**Comment by jfurcean** [comment]
This should be fixed by PR #74.
